code/modules/events/tram_malfunction.dm

```diff
@@ -0,0 +1,77 @@
+#define TRAM_MALFUNCTION_TIME_UPPER 420
+#define TRAM_MALFUNCTION_TIME_LOWER 240
+#define XING_STATE_GREEN 0
+
+/datum/round_event_control/tram_malfunction
+	name = "Tram Malfunction"
+	typepath = /datum/round_event/tram_malfunction
+	weight = 20
+	max_occurrences = 4
+	category = EVENT_CATEGORY_ENGINEERING
+
+// If manually triggered, check if there's a tram.
+/datum/round_event_control/tram_malfunction/admin_setup()
+	for(var/tram_id in GLOB.active_lifts_by_type)
+		var/datum/lift_master/tram_ref = GLOB.active_lifts_by_type[tram_id][1]
+		if(!istype(tram_ref))
+			return ADMIN_CANCEL_EVENT
+
+		if(tram_ref.specific_lift_id == MAIN_STATION_TRAM)
+			return
+
+	return ADMIN_CANCEL_EVENT
+
+// Check if there's a tram we can cause to malfunction.
+/datum/round_event_control/tram_malfunction/can_spawn_event()
+	. = ..()
+	if(!.)
+		return .
+	for(var/tram_id in GLOB.active_lifts_by_type)
+		var/datum/lift_master/tram_ref = GLOB.active_lifts_by_type[tram_id][1]
+		if(!istype(tram_ref))
+			return FALSE
```

Comment:
dont you want continue in these for loops?

Reply:
also when would the tram ref not be a lift master

Reply:
active_lifts_by_type can have lifts that aren't a tram.

Reply:
Yeah, right on both. Changed.